holopy/core/metadata.py

```diff
@@ -142,123 +45,112 @@ def __init__(self):
-        schema = Schema(positions=schema, optics=Optics())
-
-    if optics is None:
-        if isinstance(schema.optics, Optics):
-            optics = schema.optics
+def to_vector(c):
+    if c is None:
+        return c
+    if hasattr(c, 'vector'):
+        return c
+    c = np.array(c)
+    if c.shape == (2,):
+        c = np.append(c, 0)
+
+    return xr.DataArray(c, coords={vector: ['x', 'y', 'z']})
+
+def to_spherical(a, origin, wavevec=None, include_r=True):
+    xo, yo, zo = origin
+    x, y, z = a.x - xo, a.y - yo, zo - a.z
+    theta = arctan2(np.sqrt(x**2 + y**2), z)
+    phi = arctan2(y, x)
+    phi = phi + 2*np.pi * (phi < 0)
+    if include_r:
+        r = sqrt(x**2 + y**2 + z**2)
+        if wavevec is not None:
+            rname = 'kr'
+            kr = r*wavevec
         else:
-            optics = Optics()
-    optics = optics.like_me(wavelen=wavelen, index=index, polarization=polarization)
-    return schema.like_me(optics=optics)
-
+            rname = 'r'
+            kr = r
+        return xr.DataArray(a, coords={rname: kr, 'theta': theta, 'phi': phi, 'x': a.x, 'y':a.y})
+    else:
+        return xr.DataArray(a, coords={'theta': theta, 'phi': phi, 'x': a.x, 'y': a.y})
+
+def flat(a):
+    return a.stack(flat=['x', 'y'])
+
+def from_flat(a):
+    if hasattr(a, 'flat'):
+        return a.unstack('flat')
+    return a
+
+def r_theta_phi_flat(a, origin):
+    f = flat(to_spherical(a, origin))
+    return f
+    return np.vstack((f.r, f.theta, f.phi)).T
+
+def kr_theta_phi_flat(a, origin, wavevec=None):
+    if wavevec is None:
+        wavevec = get_wavevec(a)
+    return flat(to_spherical(a, origin, wavevec))
+    pos = r_theta_phi_flat(a, origin)
+    pos['r'] *= wavevec
+    return pos
+
+def theta_phi_flat(a, origin):
+    if hasattr(a, 'theta') and hasattr(a, 'phi'):
+        return a
+    return flat(to_spherical(a, origin))
+
+def make_coords(shape, spacing, z=0):
+    if np.isscalar(shape):
+        shape = np.repeat(shape, 2)
+    if np.isscalar(spacing):
+        spacing = np.repeat(spacing, 2)
+    return {'x': np.arange(shape[0])*spacing[0], 'y': np.arange(shape[1])*spacing[1], 'z': 0}
+
+def make_attrs(medium_index, illum_wavelen, illum_polarization, normals):
+    return {'medium_index': medium_index, 'illum_wavelen': illum_wavelen, 'illum_polarization': to_vector(illum_polarization), 'normals': to_vector(normals)}
+
+def angles_list(theta, phi, medium_index, illum_wavelen, illum_polarization, normals=(0, 0, 1)):
+    # This is a hack that gets the data into a format that we can use
+    # elsewhere, but feels like an abuse of xarray, it would be nice to replace this with something more ideomatic
+    d = make_attrs(medium_index, illum_wavelen, illum_polarization, normals)
+    d['theta'] = theta
+    d['phi'] = phi
+    return xr.DataArray(np.zeros(len(theta)), dims=['point'], attrs=d)
+
+def ImageSchema(shape, spacing, medium_index=None, illum_wavelen=None, illum_polarization=None, normals=(0, 0, 1)):
```

Comment:
Obviously that won't work because we need information about image size. I still think it would be nice to get rid of ImageSchema, but I can't think of a smart way to do it right now.